src/praisonai/praisonai/scheduler/agent_scheduler.py

```
@@ -242,21 +246,12 @@ def _execute_with_retry(self, max_retries: int):
-                        result = self._executor.execute(self.task)
-                        signal.alarm(0)  # Cancel alarm
-                    except AttributeError:
-                        # Windows doesn't support SIGALRM, use threading.Timer fallback
-                        logger.warning("Timeout not supported on this platform, executing without timeout")
-                        result = self._executor.execute(self.task)
+                    with ThreadPoolExecutor(max_workers=1) as executor:
+                        future = executor.submit(self._executor.execute, self.task)
+                        try:
+                            result = future.result(timeout=self.timeout)
+                        except FuturesTimeout:
+                            raise TimeoutError(f"Execution exceeded {self.timeout}s timeout")
```

Ruff (0.15.13)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)